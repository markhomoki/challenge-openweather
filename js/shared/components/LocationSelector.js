// @flow

import * as React from 'react';
import FormFieldLabel from './FormFieldLabel';

type Props = {
	label?: string,
	onFetch: Function,
};

type State = {
	value: string,
};

export default class LocationSelector extends React.PureComponent<Props, State> {

	static defaultProps = {
		label: 'Select Location',
	}

	options: Array<{|value: string, label: string|}> = [
		{ value: 'London,GB', label: 'London' },
		{ value: 'Paris,FR', label: 'Paris' },
		{ value: 'Barcelona,ES', label: 'Barcelona' },
		{ value: 'Rome,IT', label: 'Rome' },
	];

	state = {
		value: 'London,GB',
	};

	handleFetch = () => {
		this.props.onFetch(this.state.value);
	}

	handleChange = (event: SyntheticEvent<*>) => {
		const { value } = event.currentTarget;
		this.setState({ value });
	}

	render() {
		const {
			label,
		} = this.props;

		return (
			<div className="form-field-select">
				{!!label && <FormFieldLabel>{label}</FormFieldLabel>}
				<select onChange={this.handleChange} value={this.state.value}>
					{this.options.map((o) => {
						return <option key={o.value} value={o.value}>{o.label}</option>;
					})}
				</select>
				<button className="button" type="button" onClick={this.handleFetch}>Fetch</button>
			</div>
		);
	}

}
