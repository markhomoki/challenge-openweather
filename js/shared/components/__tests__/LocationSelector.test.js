import React from 'react';
import renderer from 'react-test-renderer';
import LocationSelector from '../LocationSelector';

describe('LocationSelector', () => {
	it('renders correctly', () => {
		const tree = renderer.create(<LocationSelector />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('has different label', () => {
		const wrapper = mount(<LocationSelector label="Choose Location" />);
		expect(wrapper.find('.form-field-label').text()).toEqual('Choose Location');
	});

	it('defaults to London', () => {
		const wrapper = mount(<LocationSelector />);
		expect(wrapper.state().value).toEqual('London,GB');
	});

	it('changes state on select option', () => {
		const wrapper = mount(<LocationSelector />);
		expect(wrapper.state().value).toEqual('London,GB');
		wrapper.instance().handleChange({ currentTarget: { value: 'Barcelona,ES' } });
		expect(wrapper.state().value).toEqual('Barcelona,ES');
		wrapper.instance().handleChange({ currentTarget: { value: 'Rome,IT' } });
		expect(wrapper.state().value).toEqual('Rome,IT');
	});
});
