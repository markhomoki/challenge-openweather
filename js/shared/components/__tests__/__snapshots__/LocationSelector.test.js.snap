// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationSelector renders correctly 1`] = `
<div
  className="form-field-select"
>
  <span
    className="form-field-label"
  >
    Select Location
  </span>
  <select
    onChange={[Function]}
    value="London,GB"
  >
    <option
      value="London,GB"
    >
      London
    </option>
    <option
      value="Paris,FR"
    >
      Paris
    </option>
    <option
      value="Barcelona,ES"
    >
      Barcelona
    </option>
    <option
      value="Rome,IT"
    >
      Rome
    </option>
  </select>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    Fetch
  </button>
</div>
`;
